doc/releases/changelog-0.42.0.md

```diff
@@ -4,98 +4,138 @@
+  qml.decomposition.enable_graph()
+  ```
+
+  To demonstrate the resource-efficiency of this new decomposition, let's use 
+  :func:`~.transforms.decompose` to decompose an instance of :class:`~.Select` using the new unary iterator decomposition rule, and further decompose these gates into the Clifford+T gate set
+  using :func:`~.clifford_t_decomposition` so that we can count the amount of `T` gates required:
   
-      return params
+  ```python
+  reg = qml.registers({"targ": 2, "control": 2, "work": 1})
+  targ, control, work = (reg[k] for k in reg.keys())
+
+  dev = qml.device('default.qubit')
+  ops = [qml.X(targ[0]), qml.X(targ[1]), qml.Y(targ[0]), qml.SWAP(targ)]
+
+  @qml.clifford_t_decomposition
+  @partial(qml.transforms.decompose, gate_set={
+          qml.X, qml.CNOT, qml.TemporaryAND, "Adjoint(TemporaryAND)", "CY", "CSWAP"
+      }
+  )
+  @qml.qnode(dev)
+  def circuit():
+      qml.Select(ops, control=control, work_wires=work)
+      return qml.state()
   ```
 
   ```pycon
-  >>> workflow()
-  Array([ 3.14159265, -1.57079633], dtype=float64)
+  >>> unary_specs = qml.specs(circuit)()
+  >>> print(unary_specs['resources'].gate_types["T"])
+  16
+  >>> print(unary_specs['resources'].gate_types["Adjoint(T)"])
+  13
   ```
 
-<h4>State-of-the-art templates and decompositions 🐝</h4>
+  In contract, when the unary iterator decomposition is not used (when the graph-based decomposition
+  system is disabled), and `Select` is decomposed into Clifford+T gates, we end up with orders of 
+  magnitude more `T` gates:
 
-* The decompositions of `SingleExcitation`, `SingleExcitationMinus` and `SingleExcitationPlus`
-  have been reduced to fewer rotations and/or (CNOT|CZ|CY) gates. This leads to lower circuit cost
-  when decomposing these gates, both when focusing on two-qubit gates or on non-Clifford gates.
-  [(#7771)](https://github.com/PennyLaneAI/pennylane/pull/7771)
+  ```python
+  qml.decomposition.disable_graph()
 
-* A new decomposition based on *unary iteration* has been added to :class:`qml.Select`.
-  This decomposition reduces the :class:`T` count significantly, and uses :math:`c-1`
-  auxiliary wires for a :class:`qml.Select` operation with :math:`c` control wires.
-  Unary iteration leverages these auxiliary wires to store intermediate values for reuse
-  among the different multi-controlled operators, avoiding unnecessary recomputation.
-  Check out the documentation for a thorough explanation.
-  [(#7623)](https://github.com/PennyLaneAI/pennylane/pull/7623)
-  [(#7744)](https://github.com/PennyLaneAI/pennylane/pull/7744)
+  @qml.clifford_t_decomposition
+  @qml.qnode(dev)
+  def circuit():
+      qml.Select(ops, control=control, work_wires=work)
+      return qml.state()
+  ```
+
+  ```pycon
+  >>> no_unary_specs = qml.specs(circuit)()
+  >>> print(no_unary_specs['resources'].gate_types["T"])
+  108134
+  >>> print(no_unary_specs['resources'].gate_types["Adjoint(T)"])
+  108109
```

Comment:
Those are outrageous numbers :sweat_smile: We should be careful not to point too much to the fact that Clifford+T decomposition in general is not handled very well in PL yet :grimacing:

Reply:
Lollllll